### PR TITLE
CVE-2016-6817

### PR DESCRIPTION
**Pull request by axt1384**
Researched CVE and updated yaml file

### CODE REVIEW

File: cves/CVE-2016-6817.yml [hunk 1]
```diff
@@ -143,8 +143,8 @@ interesting_commits:
     section by explaining what happened between the VCCs and the fix.
   answer:
   commits:
-    - commit:
-      note:
+    - commit: 0cb56058ee7f237764cc4435a429d366e6b7a128
+      note: This commit was about writing unit tests for the HTTP2 pings. I found it interesting because it was added out of nowhere, which made me think why they decided to add unit tests for something out of the blue. Also, I noticed that the HTTP2 header parser also didn't have unit tests, until the vulnerability was found, which I thought was interesting.
```
Comment (justinschneider1024) [inline cves/CVE-2016-6817.yml]: `Also, I noticed that the HTTP2 header parser also didn't have unit tests, until the vulnerability was found, which I thought was interesting.`
Avoid using 'also' twice. 
This sentence feels like it runs on a bit long. Try rewording into 2 sentences.

File: cves/CVE-2016-6817.yml [hunk 2]
```diff
@@ -204,4 +204,10 @@ mistakes:
...
     every one. Write a thoughtful entry here that those ing the software
     engineering industry would find interesting.
-  answer:
+  answer: 
+  In my opinion, this was a coding mistake. This specific vulnerability is based off an edge-case that, at the time, 
+  developers did not think would ever occur. The founder of the vulnerability, barney2k7, discovered 
+  such edge case when trying to parse a header about 30kb. The fix of this vulnerability involves 
+  increasing the buffer size if the capacity were to be less than the size of the header. The CWE entry for this 
+  has mitigations which the fix follows. The fix checks to make sure if the size is larger than the capacity 
+  and if it is, increase the capacity.
```
Comment (justinschneider1024) [inline cves/CVE-2016-6817.yml]: `parse a header about 30kb.`
Insert 'of' to make `parse a header of about 30kb`

File: cves/CVE-2016-6817.yml [hunk 3]
```diff
@@ -10,27 +10,27 @@ CWE_instructions: |
   Please go to cwe.mitre.org and find the most specific, appropriate CWE entry
   that describes your vulnerability. (Tip: this may not be a good one to start
   with - spend time understanding this vulnerability before making your choice!)
-CWE:
+CWE: CWE-119
```
Comment (justinschneider1024) [inline cves/CVE-2016-6817.yml:14]: I will give this CVE a 6

File: cves/CVE-2016-6817.yml [hunk 4]
```diff
@@ -43,15 +43,17 @@ description_instructions: |
   that outsiders to Chromium would not understand. Technology like "regular
   expressions" is fine, and security phrases like "invalid write" are fine to
   keep too.
-description:
+description: |
+  The HTTP/2 header parser was found to have entered an infinite loop if a header
```
Comment (cjr8920) [inline cves/CVE-2016-6817.yml:47]: Good description.

File: cves/CVE-2016-6817.yml (hunk 3 above)
Comment (cjr8920) [inline cves/CVE-2016-6817.yml:14]: I give this CVE 3 upvotes because the way it was found is interesting.

Comment (justinschneider1024) [reply cves/CVE-2016-6817.yml:14]: im going to change this to a 3

**Pull request merged**